integration-tests/kafka/src/main/java/io/quarkus/it/kafka/streams/KafkaStreamsEndpoint.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.it.kafka.streams;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.errors.InvalidStateStoreException;
+import org.apache.kafka.streams.state.QueryableStoreTypes;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+
+@ApplicationScoped
+@Path("/kafkastreams")
+public class KafkaStreamsEndpoint {
+
+    @Inject
+    KafkaStreams streams;
+
+    private ReadOnlyKeyValueStore<Integer, Long> getCountstore() {
+        while (true) {
+            try {
+                return streams.store("countstore", QueryableStoreTypes.keyValueStore());
```

Comment:
Hmmm. I see you remove awaitility but I would really prefer we use that rather than an unbounded loop.

Reply:
This one didn't use Awaility before, it's just extracted from the previous common class for pipeline and endpoint.